src/lib/Form.tsx

```diff
@@ -1,34 +1,71 @@
-import { ReactNode } from "react";
+import { ReactNode, useCallback, useEffect, useRef } from "react";
 import { DeepPartial, FieldValues, FormProvider, Resolver, SubmitHandler, useForm, UseFormReturn } from "react-hook-form";
 import { jsonIsoDateReviver } from "./helpers/dateUtils";
 import { InternalFormContext } from "./context/InternalFormContext";
+import { useDebouncedCallback } from "use-debounce";
 
 interface FormProps<T extends FieldValues> {
   onSubmit: SubmitHandler<T>;
   resolver?: Resolver<T>;
   defaultValues?: DeepPartial<T>;
   requiredFields?: (keyof T)[];
+  autoSubmitOnChangeDelayInMs?: number;
   children: ((formMethods: UseFormReturn<T, unknown>) => ReactNode) | ReactNode;
 }
 
-const Form = <T extends FieldValues>({ children, onSubmit, resolver, defaultValues, requiredFields }: FormProps<T>) => {
+const Form = <T extends FieldValues>({
+  children,
+  onSubmit,
+  resolver,
+  defaultValues,
+  requiredFields,
+  autoSubmitOnChangeDelayInMs,
+}: FormProps<T>) => {
   const revivedDefaultValues = defaultValues
     ? (JSON.parse(JSON.stringify(defaultValues), jsonIsoDateReviver) as DeepPartial<T>)
     : defaultValues;
 
   const formMethods = useForm<T>({ resolver, defaultValues: revivedDefaultValues });
-  const { handleSubmit } = formMethods;
+  const { handleSubmit, watch } = formMethods;
```

Comment:
idea: 

Move all this code into a custom hook called `useAutoSubmit`, which returns a function that can be passed to the onSubmit of the `form`. And I guess as parameters you could probably pass the `formMethods` + the `onSubmit` + `autoSubmitOnChangeDelayInMs`

Not sure how feasible so only an idea :-)

Reply:
what do you think about like it is now?